### PR TITLE
Added additional tests to img_filter unit test

### PR DESCRIPTION
**Pull request by Keanna-K**
- updated the error types in img_filter function to be more specific
- updated the unit test for img_filter to match the assertion testing style provided in the mds python packages documentation
- added tests for edge cases

### CODE REVIEW

File: tests/test_pyimager.py [hunk 1]
```diff
@@ -61,32 +61,44 @@ def test_reduce_dimensions():
...
+    # test that output image array is smaller or equal in size to original image array
+    img = Image.open('images/mandrill.jpg')
+    input_test = np.array(img)
+    assert output_test.shape[0] <= input_test.shape[0], 'Output image should have equal or smaller dimensions than original image.'
+    assert output_test.shape[1] <= input_test.shape[1], 'Output image should have equal or smaller dimensions than original image.'
+
+    # test that output image array dimensions when using sharpen filter
+    output_test =  img_filter('images/mandrill.jpg', filter_type = 'sharpen', strength=0.5)
+    assert output_test.shape[0] == input_test.shape[0] - 2, 'Output image has wrong width.'
+    assert output_test.shape[1] <= input_test.shape[1] - 2, 'Output image should has wrong height.'
```
Comment (zoepan00) [inline tests/test_pyimager.py]:
```suggestion
    assert output_test.shape[1] <= input_test.shape[1] - 2, 'Output image has wrong height.'
```
